mlir/test/Target/Cpp/declare_func.mlir

```diff
@@ -22,3 +22,17 @@ emitc.declare_func @array_arg
 emitc.func @array_arg(%arg0: !emitc.array<3xi32>) {
     emitc.return
 }
+
+// CHECK: void reference_scalar_arg(int32_t &[[V2:[^ ]*]]);
+emitc.declare_func @reference_scalar_arg
+// CHECK: void reference_scalar_arg(int32_t  &[[V2:[^ ]*]]) {
+emitc.func @reference_scalar_arg(%arg0: i32 {emitc.reference}) {
```

Comment:
If you add the reference attribute into the ODS for emitc.func, then we can do some pretty printing like
```
emitc.func @reference_scalar_arg(%arg0: i32&) {
```

Reply:
Assembly of function ops isn't part of the ODS, there's a [common parser for function signatures](https://github.com/Xilinx/llvm-project/blob/cfac8df98c04cd9a94b0a9247b53d77e8e500a22/mlir/lib/Interfaces/FunctionImplementation.cpp#L116).